delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTaskSuite.scala

```diff
@@ -0,0 +1,52 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.deletion
+
+import cats.effect.concurrent.Ref
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.CompositeViewsFixture
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.indexing.CompositeViewDef
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.indexing.CompositeViewDef.{ActiveViewDef, DeprecatedViewDef}
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.nxv
+import ch.epfl.bluebrain.nexus.delta.sdk.views.ViewRef
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Identity.{Anonymous, Subject}
+import ch.epfl.bluebrain.nexus.testkit.CirceLiteral
+import ch.epfl.bluebrain.nexus.testkit.bio.BioSuite
+import fs2.Stream
+import monix.bio.Task
+
+import java.util.UUID
+
+class CompositeViewsDeletionTaskSuite extends BioSuite with CirceLiteral with CompositeViewsFixture {
+
+  implicit private val anonymous: Subject = Anonymous
+
+  private val rev = 2
+
+  private val active1    = ViewRef(projectRef, nxv + "active1")
+  private val active2    = ViewRef(projectRef, nxv + "active2")
+  private val deprecated = ViewRef(projectRef, nxv + "deprecated")
+
+  private def activeView(ref: ViewRef) = ActiveViewDef(
+    ref,
+    UUID.randomUUID(),
+    rev,
+    viewValue
+  )
+
+  private val viewStream: Stream[Task, CompositeViewDef] =
+    Stream(
+      activeView(active1),
+      DeprecatedViewDef(deprecated),
+      activeView(active2)
+    )
+
+  test("Deprecate all active views for project") {
+    for {
+      deprecated   <- Ref.of[Task, Set[ViewRef]](Set.empty)
```

Comment:
Small thing, but perhaps this should be a list? Not sure if duplication would indicate an issue

Reply:
Duplication can't happen here as view ref is the identifier of the view and we only process each view only once.

Reply:
Well, the set here represents calls to the function, and you could call the function twice if you had a bug. I agree it's unlikely, but List would be more accurate than Set

Reply:
Things can only be deprecated once

Reply:
`deprecate(view, subject) >> deprecate(view, subject)` would be a bug but this test wouldn't fail

Reply:
The deprecate function in the test is idempotent just like the real one (and the other deletion tasks), so it wouldn't be a bug.